client/lib/credit-card-details/validation.js

```diff
@@ -197,31 +196,20 @@ export function validateCardDetails( cardDetails ) {
  * Retrieves the type of credit card from the specified number.
  *
  * @param {string} number - credit card number
- * @returns {string} the type of the credit card
+ * @returns {string|null} the type of the credit card
  * @see {@link http://en.wikipedia.org/wiki/Bank_card_number} for more information
  */
 export function getCreditCardType( number ) {
 	if ( number ) {
 		number = number.replace( / /g, '' );
 
-		if ( number.match( /^3[47]\d{0,13}$/ ) ) {
-			return 'amex';
-		} else if ( number.match( /^4\d{0,12}$/ ) || number.match( /^4\d{15}$/ ) ) {
-			return 'visa';
-		} else if (
-			number.match( /^5[1-5]\d{0,14}|^2(?:2(?:2[1-9]|[3-9]\d)|[3-6]\d\d|7(?:[01]\d|20))\d{0,12}$/ )
-		) {
-			//valid 2-series range: 2221 - 2720
-			//valid 5-series range: 51 - 55
-			return 'mastercard';
-		} else if (
-			number.match( /^6011\d{0,12}$/ ) ||
-			inRange( parseInt( number, 10 ), 622126, 622926 ) || // valid range is 622126-622925
-			number.match( /^64[4-9]\d{0,13}$/ ) ||
-			number.match( /^65\d{0,14}$/ )
-		) {
-			return 'discover';
+		const cardType = creditcards.card.type( number, false );
+
+		if ( typeof cardType === 'undefined' ) {
+			return null;
 		}
+
+		return cardType.toLowerCase();
```

Comment:
Tested the return with various cards and it works fine 👍 

Only one side effect I can see is the Amex icon doesn't work anymore. 

<img width="719" alt="screen shot 2018-01-28 at 1 34 53 pm" src="https://user-images.githubusercontent.com/6458278/35478419-2734caf6-0431-11e8-9b86-735eb8444f71.png">

<img width="726" alt="screen shot 2018-01-28 at 1 35 00 pm" src="https://user-images.githubusercontent.com/6458278/35478420-28d7ef00-0431-11e8-976a-ba6f01e9d3b9.png">

There's a class name allocation in [components/upgrades/credit-card-number-input/index.jsx](https://github.com/Automattic/wp-calypso/blob/1a0bad3/client/components/upgrades/credit-card-number-input/index.jsx#L19).

The [stylesheet has defined](https://github.com/Automattic/wp-calypso/blob/1a0bad3/client/components/upgrades/credit-card-number-input/style.scss#L10) `.amex` but we're getting `american express` back. 

One way to do it would be to `kebabCase` it:

```javascript
import React from 'react';
import { kebabCase } from 'lodash';
import classNames from 'classnames';

/**
 * Internal dependencies
 */
import { getCreditCardType } from 'lib/credit-card-details';
import Input from 'my-sites/domains/components/form/input';

class CreditCardNumberInput extends React.Component {
	render() {
		const classes = classNames( kebabCase( getCreditCardType( this.props.value ) ) );
		return (
			<div className="credit-card-number-input">
				<Input { ...this.props } classes={ classes } />
			</div>
		);
	}
}

```

Then update the class name:

```css
	.american-express input {
		background-image: url( '/calypso/images/upgrades/cc-amex.svg' );
	}
```

Reply:
Thanks for catching that. Unfortunately we use just `amex` in various other places, and I think it's best if I normalize the output in `getCreditCardType`